web/auth.go

```diff
@@ -529,9 +529,11 @@ func (h *loginHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if r.FormValue("backto") != "" {
 		backto := r.FormValue("backto")
 
-		// to prevent redirecting to an external URL we only set the session data when we fail to parse backto
-		_, err := url.ParseRequestURI(backto)
-		if err != nil {
+		// to prevent redirecting to an external URL we only set the session data when:
+		// 1. we fail to parse backto
+		// 2. backto does not include a hostname
+		parsed, err := url.ParseRequestURI(backto)
+		if err != nil || parsed.Hostname() == "" {
```

Comment:
Wondering if we need 1 if paths are always passing that parse. May just be happy to reduce to:

```suggestion
		// to prevent redirecting to an external URL we only set the session data when backto does not include a hostname
		parsed, err := url.ParseRequestURI(backto)
		if err != nil && parsed.Hostname() == "" {
```

Reply:
[unfamiliar with this area of the codebase]

I don't think `&& parsed.Hostname()` will work -- I think it will panic because `parsed` is nil when err != nil.

This works as `||` because the second check is short-circuted and not evaluated when err != nil

Reply:
Yeah I wrote this up quickly during a meeting but thinking I am ok with the code as is unless anyone has major complaints.